src/data/list/defs.lean

```diff
@@ -458,4 +465,37 @@ def choose (hp : ∃ a, a ∈ l ∧ p a) : α := choose_x p l hp
 
 end choose
 
+namespace func
+
+/- Definitions for using lists as finite 
+   representations of functions with domain ℕ. -/
+
+@[simp] def set (a' a : α) : list α → ℕ → list α
+| (_::as) 0     := a::as
+| []      0     := [a]
+| (h::as) (k+1) := h::(set as k)
+| []      (k+1) := a'::(set ([] : list α) k)
+
+@[simp] def get (d : α): ℕ → list α → α
+| _ []          := d
+| 0 (a::as)     := a
+| (n+1) (a::as) := get n as
+
+def equiv (a : α) (as1 as2 : list α) : Prop :=
```

Comment:
Looking at the lemmas, I don't understand the point of the argument `a`. If it is being viewed as a default extension value, then it should be supplied by `inhabited`.

Reply:
For pointwise arithmetic operations like `list.func.add` or `list.func.sub`, it is important that the default extension value is actually `0`, not just any default value. You can ensure this either directly by supplying an argument, or indirectly by adding a new instance whenever the existing default value is nonzero. I didn't really like the second idea, but maybe there are good arguments for it. Any thoughts?

Reply:
For `add` and `sub`, you can put `<0>` in for the inhabited argument. This looks a bit ugly in the lemmas, but it should defeq match the real inhabited instance in any reasonable inhabited type like `int` or `nat`. Alternatively you can use `has_zero` for all of these, at the risk of assuming some arithmetic structure in unrelated contexts.

Reply:
fixed